tests/unit/test_label_parsing.py

```diff
@@ -0,0 +1,330 @@
+    assert parser("a") == [ErrorTokenResult], "Should not parse the string"
+    # test that it pulls the correct number of characters when there are exactly the number it needs
+    assert parser("ab") == [
+        TokenResult("ab", Identifier("ab"), 2)
+    ], "Should parse the string"
+
+
+def test_regex_parser():
+    parser = regex(r"[/\-:_]+", Delimiter)
+    # test pulling just 1 character
+    assert parser("-abc") == [
+        TokenResult("-", Delimiter("-"), 1)
+    ], "Should parse the string"
+    # test pulling multiple characters
+    assert parser("::_abc") == [
+        TokenResult("::_", Delimiter("::_"), 3)
+    ], "Should parse the string"
+
+
+def test_choice_parser():
+    parser = choice(string("abc", Identifier), string("def", Identifier))
+    # test that it parses the first string
+    assert parser("abc") == [
+        TokenResult("abc", Identifier("abc"), 3)
+    ], "Should parse the string"
+    # test that it parses the second string
+    assert parser("def") == [
+        TokenResult("def", Identifier("def"), 3)
+    ], "Should parse the string"
+
+    # test parsing order
+    parser = choice(string("abc", Identifier), string("ab", Identifier))
+    # test that it parses the first string
+    assert parser("abc") == [
+        TokenResult("abc", Identifier("abc"), 3)
+    ], "Should parse the string"
+    # test that it parses the second string
+    assert parser("ab") == [
+        TokenResult("ab", Identifier("ab"), 2)
+    ], "Should parse the string"
+
+    # more complex parser
+    parser = choice(
+        string("abc", Identifier),
+        string("def", Identifier),
+        regex(r"[/\-:_]+", Delimiter),
+    )
+    # test that it parses the first string
+    assert parser("abc") == [
+        TokenResult("abc", Identifier("abc"), 3)
+    ], "Should parse the string"
+    # test that it parses the second string
+    assert parser("def") == [
+        TokenResult("def", Identifier("def"), 3)
+    ], "Should parse the string"
+    # test the regex matches
+    assert parser("-abc") == [
+        TokenResult("-", Delimiter("-"), 1)
+    ], "Should parse the string"
+
+
+def test_constant_parser():
+    parser = constant(Constant(BRICK.Air_Handling_Unit))
+    # test that the constant is emitted
+    assert parser("abc") == [
+        TokenResult(None, Constant(BRICK.Air_Handling_Unit), 0)
+    ], "Should parse the string"
+    assert parser("") == [
+        TokenResult(None, Constant(BRICK.Air_Handling_Unit), 0)
+    ], "Should parse the string"
+
+
+def test_abbreviations():
+    parser = abbreviations(COMMON_EQUIP_ABBREVIATIONS_BRICK)
+    # test that the constant is emitted
+    assert parser("AHU") == [
+        TokenResult("AHU", Constant(BRICK.Air_Handling_Unit), 3)
+    ], "Should parse the string"
+    # test that only the matching characters are consumed
+    assert parser("AHU1") == [
+        TokenResult("AHU", Constant(BRICK.Air_Handling_Unit), 3)
+    ], "Should parse the string"
```

Comment:
this could also possibly use a negative test case

Reply:
Added!